alphagov/paas-cf
# [#168427527] - use credhub for cf-secrets

What
----

Currently we store a bunch of database passwords in S3 in an object called cf-secrets.yml. This PR takes these secrets, writes them to credhub instead and updates the pipeline to read the secrets from credhub. If the secrets don't exist in the S3 object they'll be generated with `credhub generate --no-overwrite`, which supports the case where we're bringing up a new cloud foundry.

When I looked into the rotation process for these secrets I found that they're all skipped. This is because rotating database passwords in a zero-downtime way is tricky, so over they years we've added all of them to the list of creds that we don't rotate. Since this doesn't work anyway I've just removed the rotation code.

All the old secrets generation code is deleted because it's not used anymore 🎉 

How to review
-------------

* Code review
* Make sure you've run create-bosh-concourse in your dev env from (near) paas-bootstrap@master
* Run this branch down your dev env and observe that `credhub find -p /` now contains the cf-secrets (I've already done this in my dev env)
* Run the destroy-cloud-foundry pipeline in your dev env and then re-run the create-cloudfoundry pipeline (I've already done this in my dev env)
* Merge this PR and wait for it to deploy in staging / prod
* Review #2084 - the follow up PR to fully remove the S3 version of cf-secrets and fix the destroy-cloudfoundry pipeline

Who can review
--------------

Not @richardTowers
